diff --git a/semantic_connector/src/Entity/SemanticConnectorPPServerConnection.php b/semantic_connector/src/Entity/SemanticConnectorPPServerConnection.php
index 92c480db4..1aa47df1e 100644
--- a/semantic_connector/src/Entity/SemanticConnectorPPServerConnection.php
+++ b/semantic_connector/src/Entity/SemanticConnectorPPServerConnection.php
@@ -122,10 +122,14 @@ public function save() {
 
     // Remove SPARQL-endpoints, that do not exist anymore.
     if (!empty($sparql_endpoints_to_remove)) {
-      $connections_query = \Drupal::entityQuery('sparql_endpoint');
-      $delete_connection_ids = $connections_query->condition('url', $sparql_endpoints_to_remove, 'IN')->execute();
+      try {
+        $connections_query = \Drupal::entityQuery('sparql_endpoint');
+        $delete_connection_ids = $connections_query->condition('url', $sparql_endpoints_to_remove, 'IN')
+          ->execute();
+      }
+      catch (\Exception $e) {
 
-      SemanticConnector::deleteConnections('sparql_endpoint', $delete_connection_ids);
+      }
     }
   }
 
